optax/_src/linear_algebra.py

```diff
@@ -30,10 +32,10 @@ def global_norm(updates: base.Updates) -> base.Updates:
 
 
 def power_iteration(
-    matrix,
-    num_iters=100,
-    error_tolerance=1e-6,
-    precision=lax.Precision.HIGHEST):
+    matrix: chex.Array,
+    num_iters: Optional[int]=100,
```

Comment:
`num_iters` has a default value but I don't think it has `Optional` type. In particular, `Optional[T]` is just `Union[None, T]` and I think passing None to `num_iters` would lead to an error in line 61: `jnp.logical_and(i < num_iters, run_step)`.